indexing-service/src/main/java/io/druid/indexing/firehose/IngestSegmentFirehoseFactory.java

```diff
@@ -164,12 +165,20 @@ public Firehose connect(InputRowParser inputRowParser) throws IOException, Parse
       List<String> dims;
       if (dimensions != null) {
         dims = dimensions;
+      } else if (inputRowParser.getParseSpec().getDimensionsSpec().hasCustomDimensions()) {
+        dims = inputRowParser.getParseSpec().getDimensionsSpec().getDimensions();
```

Comment:
"feed" is a metamarkets-specific thing that we actually haven't needed for a long time. We should probably remove it from Druid :)